test/controllers/api/v2/compute_resources_controller_test.rb

```diff
@@ -476,4 +476,41 @@ def teardown
       assert_response :unprocessable_entity, "Can create libvirt compute resource with invalid name"
     end
   end
+
+  context 'ovirt' do
+    setup do
+      @simple_json = {
+        name: "oVirtTest",
+        provider: "ovirt",
+        datacenter: "a221bee8-d53b-438e-8c43-fc9dc2b770b3",
+        url: "https://localhost/ovirt-engine/api",
+        user: "admin@internal",
+        password: "changeme"
+      }
+    end
+
+    test 'Should create with custom public key' do
+      pki = <<EOJ
+-----BEGIN CERTIFICATE-----
+MIICVTCCAb6gAwIBAgIJAI1/o0diiAcSMA0GCSqGSIb3DQEBCwUAMEIxCzAJBgNV
+BAYTAlhYMRUwEwYDVQQHDAxEZWZhdWx0IENpdHkxHDAaBgNVBAoME0RlZmF1bHQg
+Q29tcGFueSBMdGQwHhcNMTgwNzI2MDYyMDM1WhcNMjMwNzI1MDYyMDM1WjBCMQsw
+CQYDVQQGEwJYWDEVMBMGA1UEBwwMRGVmYXVsdCBDaXR5MRwwGgYDVQQKDBNEZWZh
+dWx0IENvbXBhbnkgTHRkMIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCnlzrd
+Q7/+tJ3UcUo9EiTUQJkOeSo2wAHL7X8UwsXdS/vc0uGIxXwBJsFV9vWia5giki/X
+gi4fua5BxqH/qvOEEz34doS1jkGiYJeukH04EYZI6CrT6Z4UoKXbf0M46WVL1yC8
+Vbg3KdpUMAKl7OS7rgJzMTtZC2osmivx8ny0mQIDAQABo1MwUTAdBgNVHQ4EFgQU
+zbQwDxa6Oh/jLKJO9UaHUHk2Wu8wHwYDVR0jBBgwFoAUzbQwDxa6Oh/jLKJO9UaH
+UHk2Wu8wDwYDVR0TAQH/BAUwAwEB/zANBgkqhkiG9w0BAQsFAAOBgQBoRWPQUe87
+/1QWiNlQgdmYJ8mLXb9aEe4Z/I7sBTHDXS8VSe0zl47SKXuCJwWtbMfqD/3i/b68
+4tkD3XT9Tp9Da47HTRhO7HyqcF4qW41QhdJ24oTKcFBs0trMXTyjJ7Ruj+dWmCmq
+QCBdrmNXLTY2s43UMb4DLYBD/YM2M/XJmw==
+-----END CERTIFICATE-----
+EOJ
+      json = @simple_json.clone
+      json[:public_key] = pki
```

Comment:
Unless it is required for the test, as in the test is to verify a key in this format is accept, could it only be an arbitrary string and added directly in `simple_json`. Then there would be no need for cloning and setting `:public_key`

Reply:
nop, the code actually looks for the structure of the PKI and will report an error if it is not in that structure

Reply:
If you want to test that `ca_cert_store` accepts this then, this could go into `ovirt_test.rb` and here you then can stub the method. Besides asserting the response you might also want to verify that `public_key` is actually present in the `response.body`

Reply:
If I understand you correctly, it will miss part of this PR.
I want to see that it's a valid public cert, and that I'm able to work with it.
Bypassing this will render this test not useful imo.

The idea is to provide a custom public cert, and see that the API knows how to handle it properly.

Regarding the body, I agree, and I'll add support for it.

Reply:
If `ca_cert_store` is responsible for accepting and validating the public cert then test should be in `ovirt_test.rb`. An example is already there with the "accepts multiple certificates" test. 

Here you'd only want assert that a public key attribute/parameter is accepted over the API for the compute resource object and added correctly, by looking for it in the response, for this it would not be necessary to use a valid certificate and you'd not want to make this test about the functionality of a different object. 

There would basically be two tests in this PR then, which only assert functionality in the class it is testing.

Reply:
I'm not sure that I understand what exactly I should test here, do you mean to create a stub method that just return the input it gets and test to see that it got it back?

And what is the other test then, if the ovirt part already test the store?